public/app/features/provisioning/Wizard/BootstrapStep.tsx

```diff
@@ -36,180 +39,50 @@ export function BootstrapStep({ onOptionSelect, onStepUpdate }: Props) {
     formState: { errors },
   } = useFormContext<WizardFormData>();
 
-  const settingsQuery = useGetFrontendSettingsQuery();
-  const resourceStats = useGetResourceStatsQuery();
-  const currentRepoName = watch('repositoryName');
   const selectedTarget = watch('repository.sync.target');
-  const [selectedOption, setSelectedOption] = useState<ModeOption | null>(null);
-  const [hasInitialized, setHasInitialized] = useState(false);
-
-  // Get repository files count
-  const { data: filesData, isLoading: isLoadingFiles } = useGetRepositoryFilesQuery(
-    { name: currentRepoName || '' },
-    { skip: !currentRepoName }
-  );
+  const repoType = watch('repository.type');
 
-  const fileCount = filesData?.items?.length || 0;
-
-  // showing the counts seems more intresting
-  const resourceCount = useMemo(() => {
-    let count = 0;
-    if (!resourceStats.data?.instance) {
-      return count;
-    }
-    for (const x of resourceStats.data.instance) {
-      count += x.count;
-    }
-    return count;
-  }, [resourceStats.data]);
+  const resourceStats = useGetResourceStatsQuery();
+  const filesQuery = useGetRepositoryFilesQuery({ name: repoName });
+  const [selectedOption, setSelectedOption] = useState<ModeOption | null>(null);
 
-  // Check for other repositories excluding the current one
-  const [otherInstanceConnected, otherFolderConnected] = useMemo(() => {
-    if (!settingsQuery.data) {
-      return [false, false];
+  const state = useMemo(() => {
+    const v = getState(repoName, settingsData, filesQuery.data, resourceStats.data);
+    if (v.actions.length) {
+      const first = v.actions[0];
+      setSelectedOption(first);
+      onOptionSelect(first.operation === 'migrate');
+      setValue('repository.sync.target', first.target);
     }
+    return v;
+  }, [repoName, settingsData, resourceStats.data, filesQuery.data, setValue, onOptionSelect, setSelectedOption]);
```

Comment:
```suggestion
  const state = useMemo(() => {
    return getState(repoName, settingsData, filesQuery.data, resourceStats.data);
  }, [repoName, settingsData, resourceStats.data, filesQuery.data]);

  useEffect(() => {
    if (state.actions.length && !selectedOption) {
      const first = state.actions[0];
      setSelectedOption(first);
      onOptionSelect(first.operation === 'migrate');
      setValue('repository.sync.target', first.target);
    }
  }, [state, selectedOption, setValue, onOptionSelect]);
```
Let's try this, not ideal, but better have all side effects in `useEffect`.  Also not a fan of the `state` and `getState` names, it usually means data that is tracked by React and causes UI updates. Maybe rename it to system data?